python/dlisio/__init__.py

```diff
@@ -748,58 +748,47 @@ def load(path):
     """
     path = str(path)
 
-    mmap = core.mmap_source()
-    mmap.map(path)
-
     stream = open(path)
     try:
         sulpos = core.findsul(stream)
         vrlpos = core.findvrl(stream, sulpos + 80)
         sul = stream.get(bytearray(80), sulpos, 80)
-        tells, residuals, explicits = core.findoffsets(mmap, vrlpos)
     except:
         stream.close()
-        mmap.unmap()
         raise
 
-    exi = [i for i, explicit in enumerate(explicits) if explicit != 0]
 
-    try:
-        stream.reindex(tells, residuals)
-        records = stream.extract(exi)
-        stream.close()
-    except:
-        stream.close()
-        mmap.unmap()
-        raise
+    stream = core.open(path, zero=vrlpos, rp66=True)
+    lfs = []
+    while True:
+        #TODO try catch on truncated files
+        explicits, implicits = core.findoffsets(stream)
+
+        # findoffsets leaves the stream's tell at the first byte after the last
```

Comment:
> You can also move all of this "tell" counting to right before you create new stream. It would be more clear they are related. Now they are separated by too many lines of code.

Nope, the lines in-between, in particular those who do `extract` and `findfdata` both uses stream, and alteres the tell.